src/store/store.ts

```diff
@@ -0,0 +1,91 @@
+import AsyncStorage, {AsyncStorageStatic} from '@react-native-async-storage/async-storage'
+import {Subject} from 'rxjs'
+
+import {PendingTransactions} from '../types/transactions'
+
+export enum StoreKey {
+  PendingTx = 'pendingTx',
+}
+
+export type VersionableData<T> = {
+  version: number
+  data: T
+}
+
+export type PendingTxStore = VersionableData<PendingTransactions>
+
+export type StoreData<T> = T extends StoreKey.PendingTx ? PendingTxStore : never
+
+export enum StoreMethod {
+  Write,
+  Remove,
+}
+
+export type StoreEvent = {
+  key: StoreKey
+} & (
+  | {
+      value: StoreData<StoreKey>
+      method: StoreMethod.Write
+    }
+  | {method: StoreMethod.Remove}
+)
+
+export class StoreService {
+  private readonly _storage: AsyncStorageStatic
+  private readonly _prefix: string
+  private readonly _emitter$: Subject<StoreEvent>
+
+  get publisher$() {
+    return this._emitter$
+  }
+
+  constructor(prefix: string) {
```

Comment:
instead of passing the prefix in, this should be initialized with an implementation with the prefix already included
```
const wallet = await createWallet()
new Storage({
  read: (path: string) => RNAS.getItem(`${wallet.id}/${path}`),
  write: (path: string, data: unknown) => RNAS.setItem(`${wallet.id}/${path}`, data)
  ...
})
```